walkeeper/src/receive_wal.rs

```diff
@@ -293,7 +317,10 @@ impl ReceiveWalConn {
             let mut sync_control_file = false;
 
             /* Receive message header */
-            let req = self.read_req::<SafeKeeperRequest>()?;
+            let msg_bytes = self.read_msg_bytes()?;
+            let mut msg_reader = msg_bytes.reader();
+
+            let req = SafeKeeperRequest::des_from(&mut msg_reader)?;
```

Comment:
The reason I wrote it this way is so that the `SafeKeeperRequest` and the actual contents of the WAL are read from the same CopyData message -- using `read_msg` would require them to be split across two messages (which may be better anyways; I just happened to do it this way).

Reply:
After trying out splitting the messages, I ended up keeping the two parts as a single CopyData message here -- it's substantially easier on the walproposer side to do it this way